cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkTransactionService.java

```diff
@@ -258,6 +288,8 @@ public Transaction getTransaction() {
               tx = transaction = Optional.of(txClient.startLong());
             } catch (Throwable t) {
               LOG.error("Failed to start transaction for job {}", jobId, t);
+              // Set the transaction to an absent Optional to indicate the starting of transaction failed.
+              // This will prevent future call to this method to attempt to start a transaction again
```

Comment:
oh ok, even if it fails, the absent Optional will prevent that it is attempted again. ok.